### ISSUE TITLE
Move webapp static templates to Django app(s)

### ISSUE THREAD

**Issue by brylie** [CiviWiki/OpenCiviWiki#889]
The JavaScript app, located in the `webapp` directory, contains most of the frontend templates. We want to preserve the look and feel of CiviWiki through this 2.0 transition.

Move all [static templates in the `webapp` directory](https://github.com/CiviWiki/OpenCiviWiki/tree/develop/project/webapp/templates) into Django app folders (we can assist with project structure). Convert the template syntax from Handlebars (or whatever it is currently) into Django template syntax preserving the CSS and general aesthetic.

- [ ] move all templates from `/project/webapp/templates/` into `/project/threads/templates/threads/`
- [ ] if you notice accounts-related templates, move them to `/project/accounts/templates/accounts/`
- [ ] port the Handlebars template syntax into Django template syntax (help is available)

## Goal(s)
The basic goal here is to deprecate the JavaScript SPA frontend. Instead, we will use conventional Django apps that include the relevant templates.

The SPA templates have an elegant aesthetic that we want to preserve. So, the idea is to move the template definitions into the Django application directory and convert the Handlebars syntax into the Django template syntax, preserving style and logic.

## Sub-tasks

- [ ] #914 
- [x] #915

**Comment by Uttam-Singhh** [comment]
Hello @brylie,
I would love to work on this.
It would be great if you can assign me this task and guide me more on this issue.

**Comment by brylie** [comment]
Thanks, @Uttam-Singhh!

The basic goal here is to deprecate the JavaScript SPA frontend. Instead, we will use conventional Django apps that include the relevant templates. 

The SPA templates have an elegant aesthetic that we want to preserve. So, the idea is to move the template definitions into the Django application directory and convert the Handlebars syntax into the Django template syntax, preserving style and logic.

I am available to answer any questions that arise, via this issue discussion thread.

**Comment by brylie** [comment]
@Uttam-Singhh I have added a checklist to follow in the issue description.

Please let me know if you have any difficulties along the way.

**Comment by Uttam-Singhh** [comment]
Thank you so much @brylie. I will start working on this!

**Comment by Uttam-Singhh** [comment]
Hello, @brylie I am facing an issue while running the server on my system whenever I use `python manage.py`.
I get this error please help me!
![Capture](https://user-images.githubusercontent.com/63050765/119545715-bd22cb80-bdb0-11eb-831d-15f9b6118ec2.JPG)

**Comment by brylie** [comment]
@Uttam-Singhh try again on a fresh version of the `develop` branch. I have fixed the issue you are experiencing.

E.g. do

```
git checkout develop
```

then 

```
git pull
```

Base your new branch on the fresh develop branch.

**Comment by brylie** [comment]
Note: I have added the desired app structure as well:

- `/project/threads/templates/threads` for Civi and Thread related templates
- `/project/accounts/templates/accounts/` for account-related templates (log in/out, manage profile)

**Comment by aishuo07** [comment]
is this issue open

**Comment by brylie** [comment]
@aishuo07 yes, would you like to take the issue? We can break it into small parts to make it easier to get started.

**Comment by Uttam-Singhh** [comment]
Yeah, @brylie actually it is a great idea, It will be really beneficial to newcomers like me.
Also, sorry for not working on this issue I was busy with my College Exams. Apologies!

**Comment by max1204** [comment]
Anyone still working on this issue?

**Comment by aishuo07** [comment]
> @aishuo07 yes, would you like to take the issue? We can break it into small parts to make it easier to get started.

Yes Sir

**Comment by brylie** [comment]
@aishuo07 and @max1204, I have assigned you both to this issue. Perhaps you can discuss how to work together on the task?

**Comment by brylie** [comment]
> sorry for not working on this issue

@Uttam-Singhh all good. It's easy to say yes to too many things.

**Comment by aishuo07** [comment]
Yes we can discuss the Task

**Comment by aishuo07** [comment]
> @aishuo07 and @max1204, I have assigned you both to this issue. Perhaps you can discuss how to work together on the task?

Hello max Aish here you can contact me via my LinkedIn https://www.linkedin.com/in/aish-kanodia/ and we can discuss further

**Comment by aishuo07** [comment]
can u help me with handlebars what are handlebars

**Comment by brylie** [comment]
Handlebars is a template language used in many JavaScript applications.

https://handlebarsjs.com/

When writing a web application, we need to display data. We use HTML markup to display data, but sometimes need to be able to do things like loop over a list of items or display items based on conditional logic. Template languages add a bit more capabilities to the HTML.

Django has its own template language, so we need to convert the templates in the `webapp` directory into Django templates.

https://docs.djangoproject.com/en/3.2/topics/templates/

**Comment by aishuo07** [comment]
Okay so you are saying to convert the handlebars part in web app js templates to Django compatible templates and move them to the folder /project/threads/templates/threads/   and /project/threads/templates/accounts/

**Comment by brylie** [comment]
Yes. I'm not sure whether "Handlebars" is even relevant. The goal is, as you understand, to move the templates into the Django app and use Django template syntax. 

I realize this task may be more involved than I originally thought, since a lot of the logic is in Backbone.js files. Please let me know if you encounter any roadblocks.

**Comment by max1204** [comment]
@aishuo07 @brylie I did move all the templates to the folders that were specified. And so far it hasn't broken anything. 
I still have to separate the account templates. for now, they all are in the same folder. I'll identify and then move them to `/project/accounts/templates/accounts/`

**Comment by aishuo07** [comment]
hii max how can we connect

**Comment by max1204** [comment]
@aishuo07 Let's connect on discord. My handle is `max1204#8184`

**Comment by max1204** [comment]
@brylie 
I'm testing the app after changes and there is a URL /settings the view for this is in `/project/frontend_views/views.py` it is trying to access the property `longitude` and `latitude` but the Account model doesn't have any property like that.
Someone working on it or this is still due?

**Comment by brylie** [comment]
@max1204, the user location data has been removed from the account, so should also be removed from any views, templates, etc. If you come across any user location fields, feel free to remove them as part of your PR.

**Comment by max1204** [comment]
@brylie , What do we do about the templates that don't exist. i.e. `partials/account/tabs/my_bills.html` `map.html` ?

Also if I can see the demo as there are two URLs that are not printing anything they are empty. Just wanted to make sure what changes I  need to make in order to clear the Handlebars syntax.

**Comment by brylie** [comment]
@max12304, both `my_bills` and `map` templates should no longer be needed since we have removed the legislation and user location features. Are there any other templates that don't have a home?

**Comment by max1204** [comment]
@brylie, so far these are the ones that are not available. RIght now I'm looking into the settings URL it is empty, not sure if that is empty because work is being done on this or not.

**Comment by brylie** [comment]
@max1204 just focus on the templates that you are able to find for now. Keep a list of missing templates, and we can create a follow-up task.

**Comment by manuneerav** [comment]
is this issue still open?

**Comment by brylie** [comment]
@manuneerav yes, the issue is still open. We have two developers who are taking a look. Perhaps you could coordinate with @max1204 to see how they may need assistance? This task is great for pair programming :-)

**Comment by max1204** [comment]
@manuneerav  Let's connect on discord. My handle is max1204#8184. 
My progress is a bit slow on weekdays. But I think this weekend it will be finished.

@brylie I have a few questions about some URLs.

**Comment by brylie** [comment]
@max1204 what are your questions about URLs?

**Comment by max1204** [comment]
@brylie, I have a question about the URL /settings and /profile/{username}. So on my local setup when I'm running these URLs I get the header and footer but the middle section is empty. 

Is someone working on it or it should print something?

**Comment by brylie** [comment]
@max1204 It might be better to re-write the missing template elements from scratch, using your own insight and intuition. Start by sketching what a user profile would look like (e.g. draw it on paper) and post your sketch to this thread. Then, we will create a simple Django template based on your sketch.

**Comment by brylie** [comment]
@max1204, I created separate tickets to track the user profile and settings template/view:
https://github.com/CiviWiki/OpenCiviWiki/issues/914
https://github.com/CiviWiki/OpenCiviWiki/issues/915

What other templates should I create tasks for?

@aishuo07 @manuneerav @Uttam-Singhh and @max1204 please determine if/how to divide up the tasks so they can be completed in parallel.

**Comment by max1204** [comment]
@brylie I don't there are more I'll work on it this weekend. I was busy last 2-3 days.
I don't think anyone else is working on it.

**Comment by Tanmoy07tech** [comment]
Is this issue still opened?

**Comment by brylie** [comment]
Yes. It is a large task, though. We need to create a new app called `discussions` containing templates related to Civis and threads. The project is currently undergoing some large code changes, making the work a bit more precarious.

**Comment by Theskyspace** [comment]
Do we have one more place for a newbie to pop in?

**Comment by brylie** [comment]
@Theskyspace yes. In particular, we need help finding bugs with the current development branch. Would you be willing to start a development environment and test out the app to help identify bugs?

**Comment by Theskyspace** [comment]
Yeah Sure @brylie I'll get myself hooked with his task ASAP. Let me know about particulars if any. Cheers

**Comment by brylie** [comment]
Closing this since significant progress has been made. We will follow up with a task to port Backbone templates to Django templates.
